Source/CarthageKitTests/BuildArgumentsSpec.swift

```diff
@@ -0,0 +1,327 @@
+import Quick
+import Nimble
+import CarthageKit
+
+class BuildArgumentsSpec: QuickSpec {
+	override func spec() {
+		describe("arguments") {
+			context("when created with a workspace file") {
+				let workspace = ProjectLocator.Workspace(NSURL(string: "file:///Foo/Bar/workspace.xcworkspace")!)
+
+				it("has a default set of arguments specifying the workspace") {
+					let subject = BuildArguments(project: workspace)
+
+					expect(subject.arguments) == [
+						"xcodebuild",
+						"-workspace",
+						"/Foo/Bar/workspace.xcworkspace",
+						"CODE_SIGNING_REQUIRED=NO",
+						"CODE_SIGN_IDENTITY="
+					]
+				}
+
+				it("includes the scheme if one is given") {
+					var subject = BuildArguments(project: workspace)
+					subject.scheme = "exampleScheme"
+
+					expect(subject.arguments) == [
+						"xcodebuild",
+						"-workspace",
+						"/Foo/Bar/workspace.xcworkspace",
+						"-scheme",
+						"exampleScheme",
+						"CODE_SIGNING_REQUIRED=NO",
+						"CODE_SIGN_IDENTITY="
+					]
+				}
+
+				it("includes the configuration if one is given") {
+					var subject = BuildArguments(project: workspace)
+					subject.configuration = "exampleConfiguration"
+
+					expect(subject.arguments) == [
+						"xcodebuild",
+						"-workspace",
+						"/Foo/Bar/workspace.xcworkspace",
+						"-configuration",
+						"exampleConfiguration",
+						"CODE_SIGNING_REQUIRED=NO",
+						"CODE_SIGN_IDENTITY="
+					]
+				}
+
+				describe("specifying the sdk") {
+					it("includes the sdk if it's not .MacOSX") {
+						for sdk: SDK in [.iPhoneOS, .iPhoneSimulator, .watchOS, .watchSimulator, .tvOS, .tvSimulator] {
+							var subject = BuildArguments(project: workspace)
+							subject.sdk = sdk
```

Comment:
What do you think about putting `it` inside of the loop so that each one is a separate tests? Then yo can include the `sdk` name in the test name, and failures would be more clear since each test would only have one assertion.